mongodbatlas/fw_resource_mongodbatlas_database_user.go

```diff
@@ -0,0 +1,531 @@
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"auth_database_name": schema.StringAttribute{
+				Required: true,
+			},
+			"username": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"password": schema.StringAttribute{
+				Optional:  true,
+				Sensitive: true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.Expressions{
+						path.MatchRelative().AtParent().AtName("x509_type"),
+						path.MatchRelative().AtParent().AtName("ldap_auth_type"),
+						path.MatchRelative().AtParent().AtName("aws_iam_type"),
+					}...),
+				},
+			},
+			"x509_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("NONE"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("NONE", "MANAGED", "CUSTOMER"),
+				},
+			},
+			"oidc_auth_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("NONE"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("NONE", "IDP_GROUP"),
+				},
+			},
+			"ldap_auth_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("NONE"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("NONE", "USER", "GROUP"),
+				},
+			},
+			"aws_iam_type": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("NONE"),
+				Validators: []validator.String{
+					stringvalidator.OneOf("NONE", "USER", "ROLE"),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"roles": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"collection_name": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+						"database_name": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+						"role_name": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"labels": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+						"value": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"scopes": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
+							Optional: true,
+							Computed: true,
```

Comment:
do you recall why scopes name and type attribute are now marked as computed? Can see they were originally optional: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/b1c79d09454660807682344dfa200708bfb5b8ca/mongodbatlas/resource_mongodbatlas_database_user.go#L127

Reply:
updated